app/scripts/app.js

```diff
@@ -21,14 +21,34 @@ taskList.config(["$stateProvider", "$locationProvider", function($stateProvider,
 taskList.controller("MainController.controller", ["$scope", "$firebaseArray", function($scope, $firebaseArray) {
 
   var ref = new Firebase("https://luminous-fire-9311.firebaseio.com/messages");
-  $scope.messages = $firebaseArray(ref);
+  $scope.tasks = $firebaseArray(ref);
+  var intervalID;
+
+  $scope.addTask = function() {
+    var time = new Date();
+    $scope.tasks.$add({
+      desc: $scope.newTaskDescription,
```

Comment:
It won't be a problem in that application, but as a generic piece of advice, make sure you "filter" or "sanitize" any user-provided input before registering it somewhere, especially if you intent to eventually reuse that input in the DOM (here, you'll eventually display the description in the page). A user may try to push some malicious code to your application (like JavaScript for instance).

You could [use a regular expression](http://stackoverflow.com/questions/3561493/is-there-a-regexp-escape-function-in-javascript/3561711#3561711), or maybe a [third-party library](https://www.npmjs.com/package/sanitize-caja).